media-sound/sinus/patches/sinus-0.1.3.patchset

```diff
@@ -0,0 +1,40 @@
+From 53e1d26af0b1603a1603392fc4edb476634a7b13 Mon Sep 17 00:00:00 2001
+From: Bach Nguyen <bach5000@gmail.com>
+Date: Mon, 19 Nov 2018 18:43:48 +0000
+Subject: Fix build
+
+
+diff --git a/sinus.c b/sinus.c
+index aaab435..b3deac3 100644
+--- a/sinus.c
++++ b/sinus.c
+@@ -237,7 +237,7 @@ void wave_to_little_endian(struct wave_t *wave)
+ #else
+   int i;
+   void *p;
+-
++  UINT16* piplusi;
+   switch(wave->point_size)
+   {
+     case 1:
+@@ -247,7 +247,7 @@ void wave_to_little_endian(struct wave_t *wave)
+       for(i=0; i<wave->data_size; i+=2)
+       {
+         dprintf((stderr, "host -> le: %04X ->", (UINT16)*((UINT16 *)(p+i))));
+-        (INT16)*((UINT16 *)(p+i)) = HOST_TO_LENDIAN_INT16(*((UINT16 *)(p+i)));
++        piplusi = HOST_TO_LENDIAN_INT16(*((UINT16 *)(p+i)));
```

Comment:
Depending on if HOST_TO_LENDIAN_INT16 modifies the data p+i points at, either piplusi is superfluous or the data is not converted to little endian.

Reply:
@owenca 
I have no idea what the author was trying to do here... `(INT16)*((UINT16 *)(p+i)) =`
I also have no idea why I added a variable...

Reply:
He or she was trying to do this:
```
UINT16* ptr = p; // conversion from void* to any other pointer type is allowed
ptr += i;
dprintf((stderr, "host -> le: %04X ->", *ptr);
*ptr = HOST_TO_LENDIAN_INT16(*ptr);
```

Reply:
@owenca You're right. These lines aren't needed and when removed the command line still builds.

Updating patchset as of now.

Reply:
I don't think you can take out the call to the HOST_TO_LENDIAN_INT16 macro/function (most likely a macro). Otherwise, if the data is in big endian, it won't get converted to little endian.

Reply:
@owenca That actually makes sense. I'll try building with those lines and see what happens.

Reply:
@owenca Got it to build and work (I think? I ran some examples) with those changes.
Updating the patchset once again...

Reply:
Okay... I think this should do it. Not sure if it works as intended, though, because `audio playback is not supported on this machine`...

Reply:
Have you compared the Before and After behaviors on both architectures? It will probably build on x86 now. I'd add it to the recipe and see what happens.

Reply:
@owenca Will do :)

Reply:
I suppose if an application runs on both the 32-bit and 64-bit versions, it probably will run on the hybrid too.

Reply:
On x86_64, both before and after versions of Sinus seem to be the same...
Checking gcc2...

Reply:
Ok checked on all architectures and I get the same result.
Adding "x86" to ARCH.